cdap-ui/app/directives/tooltip-toggle/tooltip-toggle.js

```diff
@@ -39,20 +39,22 @@ angular.module(PKG.name + '.commons')
     return {
       restrict: 'A',
       scope: {
-        ellipsis: '='
+        ellipsis: '=',
+        offsetValue: '='
       },
-      link: function (scope, element, attrs) {
+      link: function (scope, element) {
         function isEllipsisActive(e) {
-          return (e[0].offsetWidth > e[0].parentElement.offsetWidth - 20);
+          return (e[0].offsetWidth > e[0].parentElement.offsetWidth - scope.offsetValue);
         }
 
-        scope.$watch(attrs.tooltipEllipsis, function () {
-          if (attrs.tooltipEllipsis) {
-            if (isEllipsisActive(element)) {
-              scope.ellipsis = true;
-            } else {
-              scope.ellipsis = false;
-            }
+        scope.$watch(function () {
+          return element[0].parentElement.offsetWidth;
+        }, function () {
+          console.log('parent width: ', element[0].parentElement.offsetWidth - scope.offsetValue);
```

Comment:
We should probably remove this ```console.log``` statement. Its going to be printed a lot if we do it in ```$scope.$watch```